kubernetes/websocket/kube_exec.c

```diff
@@ -0,0 +1,122 @@
+#include <stdio.h>
+#include "kube_exec.h"
+
+#define WS_PATH_BUFFER_SIZE 1024
+#define URL_COMMAND_STRING_BUFFER_SIZE 1024
+#define COMMAND_STRING_DELIM " "
+#define COMMAND_PREFIX "&command="
+#define CONTAINER_PREFIX "&container="
+#define ESCAPED_STRING_BUFFER_SIZE 8
+#define ESCAPED_STRING_LENGTH 3
+#define TRUE_STRING "true"
+#define FALSE_STRING "false"
+
+static void default_exec_data_callback(void **p_data_received, long *p_data_received_len)
+{
+    printf("%s: Received %ld bytes:\n%s", __func__, *p_data_received_len, (char *) (*p_data_received));
+}
+
+static void escape_character_in_url(char *escaped, int escaped_buffer_size, const char chr)
```

Comment:
use `curl_easy_escape(...)` instead?

https://curl.se/libcurl/c/curl_easy_escape.html

Reply:
`escape_character_in_url` is different with `curl_easy_escape`, the former receives only one character e.g. `+` and then returns an escaped string, e.g. `%2B` at one time. But `curl_easy_escape` receives a string and returns the full escaped string for the whole url string.

At the case of `kube exec`, we need handle the characters one by one because the space (" ") cannot be escaped, it needs to be converted to "&command=", e.g. 
"ls /" will be converted to
"command=ls&command=/"